matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/keysbackup/RustKeyBackupService.kt

```diff
@@ -822,116 +799,110 @@ internal class RustKeyBackupService @Inject constructor(
+                    // If there is one, maybeBackupKeys will be called again.
+                    checkAndStartKeysBackup()
+                }
+                state == KeysBackupState.ReadyToBackUp -> {
+                    keysBackupStateManager.state = KeysBackupState.WillBackUp
+
+                    // Wait between 0 and 10 seconds, to avoid backup requests from
+                    // different clients hitting the server all at the same time when a
+                    // new key is sent
+                    val delayInMs = Random.nextLong(KEY_BACKUP_WAITING_TIME_TO_SEND_KEY_BACKUP_MILLIS)
+
+                    importScope.launch {
+                        delay(delayInMs)
+                        tryOrNull("AUTO backup failed") { backupKeys() }
                     }
                 }
-            }
-            else                                   -> {
-                Timber.v("maybeBackupKeys: Skip it because state: $state")
+                else                                   -> {
+                    Timber.v("maybeBackupKeys: Skip it because state: $state")
+                }
             }
         }
     }
 
     /**
      * Send a chunk of keys to backup
      */
-    @UiThread
     private suspend fun backupKeys(forceRecheck: Boolean = false) {
         Timber.v("backupKeys")
+        withContext(coroutineDispatchers.crypto) {
+            // Sanity check, as this method can be called after a delay, the state may have change during the delay
+            if (!isEnabled || !olmMachine.backupEnabled() || keysBackupVersion == null) {
+                Timber.v("backupKeys: Invalid configuration $isEnabled ${olmMachine.backupEnabled()} $keysBackupVersion")
+//            backupAllGroupSessionsCallback?.onFailure(IllegalStateException("Invalid configuration"))
+                resetBackupAllGroupSessionsListeners()
 
-        // Sanity check, as this method can be called after a delay, the state may have change during the delay
-        if (!isEnabled || !olmMachine.backupEnabled() || keysBackupVersion == null) {
-            Timber.v("backupKeys: Invalid configuration $isEnabled ${olmMachine.backupEnabled()} $keysBackupVersion")
-            backupAllGroupSessionsCallback?.onFailure(IllegalStateException("Invalid configuration"))
-            resetBackupAllGroupSessionsListeners()
-
-            return
-        }
+                return@withContext
+            }
 
-        if (state === KeysBackupState.BackingUp && !forceRecheck) {
-            // Do nothing if we are already backing up
-            Timber.v("backupKeys: Invalid state: $state")
-            return
-        }
+            if (state === KeysBackupState.BackingUp && !forceRecheck) {
+                // Do nothing if we are already backing up
+                Timber.v("backupKeys: Invalid state: $state")
+                return@withContext
+            }
 
-        Timber.d("BACKUP: CREATING REQUEST")
+            Timber.d("BACKUP: CREATING REQUEST")
```

Comment:
Whoops, I meant to remove those, seems that they lingered on.

We can probably just remove those ALL CAPS log lines now.